streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalProcessorContext.java

```diff
@@ -66,4 +68,15 @@
      * Mark this context as being uninitialized
      */
     void uninitialize();
+
+    /**
+     * Get a correctly typed state store, given a handle on the original builder.
+     * @param builder
+     * @param <T>
+     * @return
+     */
+    @SuppressWarnings("unchecked")
+    default <T extends StateStore> T getStateStore(final StoreBuilder<T> builder) {
+        return (T) getStateStore(builder.name());
+    }
```

Comment:
Added this so that we can have more type safety with the stores... Instead of constructing the store builder in one place and passing the name through to the processor, and then casting to (hopefully) the right store type, we can now pass the builder itself to the processors and use it as a token to get the registered store, since the builder is a witness for both the name and the correct type.

We should consider something like this in the future for the Processor API, but for now, I made it an internal-only change. I felt it was necessary to introduce this level of safety, since we create the subscription store and then use it in three different processors.